### ISSUE TITLE
Issue with Directshow and colour crunch in Unity3D 2019

### ISSUE THREAD

**Issue by phileday** [RenderHeads/UnityPlugin-AVProVideo#122]
**Describe the bug**
I've noticed that in Unity 2019 that the directshow colours for video are now crunched (appears a bit posterised).

**Your Setup (please complete the following information):**
 - Unity version: 2019
 - AVPro Video version: 1.10.0
 - Operating system version: Windows 10

**To Reproduce**
1. set to directshow and run in Unity3D 2019. All videos seem to have this.

**Screenshots**
![example](https://user-images.githubusercontent.com/16843966/61371963-e2c61c00-a88d-11e9-9a6f-118c1b4c2be9.jpg)

Hope this helps

Phil

**Comment by AndrewRH** [comment]
Yeah we have noticed that there was been some very small change in gamma/linear handling in Unity 2019...We'll investigate

**Comment by phileday** [comment]
Has there been any progress with this? I'm currently sticking with 2018 but it would be good to be able to update to the latest version of Unity3D.

**Comment by phileday** [comment]
This is still a problem in Unity3D 2020 and appears to be a problem in all Unity versions above 2018.

**Comment by AndrewRH** [comment]
hmmm sorry we haven't yet had a chance to look at this again :(

**Comment by AndrewRH** [comment]
So when I play BigBuck video on the 01_Demo_BackgroundShader demo:

1) MediaFoundation + Gamma = fine
2) MediaFoundation + Linear= fine
3) DirectShow + Gamma = fine
4) DirectShow + Linear = slightly more contrasty

It could be something with our internal blit to convert to sRGB....

Will have to see how/why it's different in 2017

Still investigating....

**Comment by phileday** [comment]
Hi
Thanks for looking into this.
You were working on a new version of your video plugin. That was put on hold I believe. Is that starting up again?
To be honest the only issue I have with your plugin at the moment is that people are reporting that Whirligig isn't playing 5k videos smoothly. I have no idea if this is my fault, Unity3D's engine or whether your plugin could improve things. I've been bating the naysayers off for now. One of the reasons I want to try the latest versions of Unity3D is to see if this improves anything. I also tend to try and keep up to date on these things as well :)
All the best and thanks for your support
Phil
Twitter - https://twitter.com/whirligigxyzBlog - http://www.whirligig.xyz/blogYoutube - https://www.youtube.com/channel/UC1MI0yclkwQjWjLw5bviXtg 

    On Friday, 21 February 2020, 15:49:04 GMT, Andrew <notifications@github.com> wrote:  
 
 
So when I play BigBuck video on the 01_Demo_BackgroundShader demo:
   
   - MediaFoundation + Gamma = fine
   - MediaFoundation + Linear= fine
   - DirectShow + Gamma = fine
   - DirectShow + Linear = slightly more contrasty

It could be something with our internal blit to convert to sRGB....

Will have to see how/why it's different in 2017

Still investigating....

**Comment by AndrewRH** [comment]
Okay we figured out the problem....it was related to how we were done some internal blitting to convert from NV12 to RGBA and whether the textures were marked as SRGB or not.  It seems that older Unity versions just considered all external textures as sRGB in linear mode, but that changed in Unity 2019...or something like that!  Not totally sure...

But we have fixed the issue and it will be included in the next update of AVPro Video :)

**Comment by phileday** [comment]
Hi Andrew 
That's brilliant to hear. I've wanted to move to the latest stable release of Unity3D but I've not been able to so this update will really help that.
Let me know if I can help in any way. I look forward to the update.
All the best
Phil
Twitter - https://twitter.com/whirligigxyzBlog - http://www.whirligig.xyz/blogYoutube - https://www.youtube.com/channel/UC1MI0yclkwQjWjLw5bviXtg 

    On Thursday, 12 March 2020, 17:18:13 GMT, Andrew <notifications@github.com> wrote:  
 
 


Okay we figured out the problem....it was related to how we were done some internal blitting to convert from NV12 to RGBA and whether the textures were marked as SRGB or not. It seems that older Unity versions just considered all external textures as sRGB in linear mode, but that changed in Unity 2019...or something like that! Not totally sure...

But we have fixed the issue and it will be included in the next update of AVPro Video :)
